### ISSUE TITLE
Error when trying to train the prior

### ISSUE THREAD

**Issue by iamzoltan** [acids-ircam/RAVE#55]
Hey again,

I just finished training and exporting a new model, but I cant seem to get it to train the prior. I am getting the following error when exporting the model:

`/home/user/code/RAVE/env3.9/lib/python3.9/site-packages/pytorch_lightning/core/saving.py:217: UserWarning: Found keys that are not in the model state dict but in the checkpoint: ['decoder.net.2.net.0.aligned.paddings.0.pad', 'decoder.net.2.net.0.aligned.paddings.1.pad', 'decoder.net.2.net.1.aligned.paddings.0.pad', 'decoder.net.2.net.1.aligned.paddings.1.pad', 'decoder.net.2.net.2.aligned.paddings.0.pad', 'decoder.net.2.net.2.aligned.paddings.1.pad', 'decoder.net.4.net.0.aligned.paddings.0.pad', 'decoder.net.4.net.0.aligned.paddings.1.pad', 'decoder.net.4.net.1.aligned.paddings.0.pad', 'decoder.net.4.net.1.aligned.paddings.1.pad', 'decoder.net.4.net.2.aligned.paddings.0.pad', 'decoder.net.4.net.2.aligned.paddings.1.pad', 'decoder.net.6.net.0.aligned.paddings.0.pad', 'decoder.net.6.net.0.aligned.paddings.1.pad', 'decoder.net.6.net.1.aligned.paddings.0.pad', 'decoder.net.6.net.1.aligned.paddings.1.pad', 'decoder.net.6.net.2.aligned.paddings.0.pad', 'decoder.net.6.net.2.aligned.paddings.1.pad', 'decoder.net.8.net.0.aligned.paddings.0.pad', 'decoder.net.8.net.0.aligned.paddings.1.pad', 'decoder.net.8.net.1.aligned.paddings.0.pad', 'decoder.net.8.net.1.aligned.paddings.1.pad', 'decoder.net.8.net.2.aligned.paddings.0.pad', 'decoder.net.8.net.2.aligned.paddings.1.pad', 'decoder.synth.paddings.0.pad', 'decoder.synth.paddings.1.pad', 'decoder.synth.paddings.2.pad']
  rank_zero_warn(
`

any ides?

**Comment by caillonantoine** [comment]
I'm working on a fix for this problem ! As a temporary workaround you can _convert_ your checkpoint using this function:

```python
import torch
import os


def convert_checkpoint(ckpt_path: str):
    """
    Remove pad buffers from a checkpoint and save
    the new converted checkpoint in the same folder
    """
    ckpt = torch.load(ckpt_path)
    keys = filter(lambda n: "pad" not in n, ckpt["state_dict"].keys())
    ckpt["state_dict"] = {k: ckpt["state_dict"][k] for k in keys}
    target = os.path.join(os.path.dirname(ckpt_path), "converted.ckpt")
    torch.save(ckpt, target)


# FOR EXAMPLE
# convert_checkpoint("runs/ljspeech/rave/version_0/checkpoints/best.ckpt")
```

 It should work for #53 too. I'll update this issue when the final fix is ready !

**Comment by iamzoltan** [comment]
thanks, ill give it a shot

**Comment by caillonantoine** [comment]
Should be fixed in d9f55f59627ea02689578d0ddae05b420be8d4d2 can you check ?

**Comment by caillonantoine** [comment]
By the way I'm closing since it's a duplicate of #45

**Comment by iamzoltan** [comment]
sounds good, ill will check shortly once this current model is done. 

It seems the smaller model successfully got to stage 2, although it looks as those the loss in increasing, is this normal? and how long should one train in stage 2?
